secure-api-gateway-ob-uk-rs-server/src/test/java/com/forgerock/sapi/gateway/ob/uk/rs/server/configuration/JacksonMapperTest.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright © 2020-2022 ForgeRock AS (obst@forgerock.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.sapi.gateway.ob.uk.rs.server.configuration;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+
+import java.math.BigDecimal;
+import java.math.MathContext;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@Slf4j
+@SpringBootTest(classes = {RsApplicationConfiguration.class})
+public class JacksonMapperTest {
+
+    @Autowired
+    private MappingJackson2HttpMessageConverter mappingJacksonHttpMessageConverter;
+
+    private ObjectMapper mapper;
+
+    @BeforeEach
+    public void setup() {
+        mapper = mappingJacksonHttpMessageConverter.getObjectMapper();
+    }
+
+    private static Stream<Arguments> deserializeArgumentsProvider() {
+        return Stream.of(
+                Arguments.arguments(
+                        "Zeros precision[2]",
+                        "1.00",
+                        new BigDecimal("1.00").setScale(BigDecimalDefaults.DEFAULT_SCALE, BigDecimalDefaults.DEFAULT_ROUNDING_MODE)
```

Comment:
Remove setScale and rounding from the expectedResult, we want this value to purely be the result with no logic attached to it. It also makes the code more readable, at a glance I can see the input and the expected output.

In this case, I assume we will get 4dp therefore the expected result is: `new BigDecimal("1.0000")`